src/rgw/driver/d4n/rgw_sal_d4n.cc

```diff
@@ -2391,6 +2410,9 @@ int D4NFilterObject::D4NFilterDeleteOp::delete_obj(const DoutPrefixProvider* dpp
   if (!source->check_head_exists_in_cache_get_oid(dpp, head_oid_in_cache, attrs, block, y) && !block.deleteMarker) {
     /* for a dirty object, if the first call is a simple delete after versioning is enabled, the call will go to the backend store and create a delete marker there
        since no object with source->get_name() will be found in the cache (and this is correct) */
+    if (source->cache_request) {
```

Comment:
at the beginning of the method, you can save the value of source->get_cache_request() in a bool variable and use that everywhere